app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tooltip.kt

```diff
@@ -158,49 +162,55 @@ private fun InnerChatTooltip(
+        }
       }
       val squircleOutline = this@withBeak.createOutline(
         squircleSize,
         layoutDirection,
         density,
       )
       val squirclePath: Path = (squircleOutline as Outline.Generic).path
-      val beakPath = when (beakDirection) {
-        BottomCenter, BottomStart, BottomEnd -> beakToBottom(density, arrowWidth)
-        TopCenter, TopStart, TopEnd -> beakToTop(density, arrowWidth)
-        Start -> beakToStart(density, arrowWidth)
-        End -> beakToEnd(density, arrowWidth)
-      }
       val squircleOffset: Offset = when (beakDirection) {
-        BottomCenter, BottomStart, BottomEnd -> Offset(x = 0f, y = -arrowHeight)
+        BottomCenter, BottomStart, BottomEnd -> Offset(x = 0f, y = 0f)
         TopCenter, TopStart, TopEnd -> Offset(x = 0f, y = arrowHeight)
         Start -> Offset(x = arrowHeight, y = 0f)
         End -> Offset(0f, y = 0f)
       }
+      val beakPath = beakToTop(density, arrowWidth).apply {
+        when (beakDirection) {
+          BottomCenter, BottomStart, BottomEnd -> transform(Matrix().apply { rotateZ(180f) })
+          TopCenter, TopStart, TopEnd -> Unit
+          Start -> transform(Matrix().apply { rotateZ(270f) })
+          End -> transform(Matrix().apply { rotateZ(90f) })
+        }
+      }
+      val offsetFromEdgeWhenOffCentered = arrowSpaceFromEdgeWhenOffCentered + (arrowWidth / 2)
       val beakOffset: Offset = when (beakDirection) {
-        BottomCenter -> Offset(x = size.width / 2, y = size.height - 2 * arrowHeight)
-        BottomStart -> Offset(x = iconWidth / 2, y = size.height - 2 * arrowHeight)
-        BottomEnd -> Offset(x = size.width - (iconWidth / 2), y = size.height - 2 * arrowHeight)
-        TopEnd -> Offset(x = size.width - (iconWidth / 2), y = arrowHeight)
-        TopCenter -> Offset(x = size.width / 2, y = arrowHeight)
-        TopStart -> Offset(x = iconWidth / 2, y = arrowHeight)
-        Start -> Offset(x = arrowHeight, y = size.height / 2)
-        End -> Offset(x = size.width - arrowHeight, y = size.height / 2)
+        BottomStart -> Offset(x = offsetFromEdgeWhenOffCentered, y = squircleSize.height)
+        BottomCenter -> Offset(x = squircleSize.width / 2, y = squircleSize.height)
+        BottomEnd -> Offset(x = squircleSize.width - offsetFromEdgeWhenOffCentered, y = squircleSize.height)
+        TopStart -> Offset(x = arrowSpaceFromEdgeWhenOffCentered, y = arrowHeight)
+        TopCenter -> Offset(x = squircleSize.width / 2, y = arrowHeight)
+        TopEnd -> Offset(x = squircleSize.width - offsetFromEdgeWhenOffCentered, y = arrowHeight)
+        Start -> Offset(x = arrowHeight, y = squircleSize.height / 2)
+        End -> Offset(x = squircleSize.width, y = squircleSize.height / 2)
       }
       return Outline.Generic(
-        Path().apply {
-          addPath(path = squirclePath, offset = squircleOffset)
-          addPath(path = beakPath, offset = beakOffset)
-        },
+        Path.combine(
```

Comment:
Could you please explain here?

Reply:
Takes two standalone paths, and does a Union on them to merge them together.
Union as in boolean algebra union, this picture might help
![image](https://github.com/user-attachments/assets/3d80b3ac-c6c2-49fd-928e-562cd480eb2e)

It simply adds both areas together. If there is any overlap, it is simply also added to the final area